code/2/2.02.constexpr.cpp

```diff
@@ -17,30 +16,31 @@ constexpr int len_foo_constexpr() {
     return 5;
 }
 
-// error in c++11
-// constexpr int fibonacci(const int n) {
-//     if(n == 1) return 1;
-//     if(n == 2) return 1;
-//     return fibonacci(n-1) + fibonacci(n-2);
-// }
-
-// ok
+// error in C++11 due to more than one return statement, okay C++14 and above
 constexpr int fibonacci(const int n) {
+     if(n == 1) return 1;
+     if(n == 2) return 1;
+     return fibonacci(n-1) + fibonacci(n-2);
+}
+
+// ok in C++11, single return
+constexpr int fibonacci2(const int n) {
     return n == 1 || n == 2 ? 1 : fibonacci(n-1) + fibonacci(n-2);
 }
 
 
 int main() {
+    const int len =10;
```

Comment:
Please kindly explain why would you like to use `const` statement to replace Marco.

Besides, you need to update the corresponding code snippet listed in the book if you making changes here, and similar to follows.

Reply:
I changed the macro constant with `const` integer value. Because macros doesn't rely on namespaces. It's better to define them _const_ or if it's length of a container _constexpr_ :+1:

Reply:
Looks like the books is typesetted using _TeX_, but my skills are a little short to edit it for now. I'll invest time to learn it in first opportunity.

Reply:
I understand your concern. However, the main body of the book has some description rely on the Marco statement and hence changes to the code also require to rephrase the paragraph in the book.

Besides, this repo has open sourced for years. It provides various ways of reading, therefore consistency of contribution seems a little bit confusing to others.
The book was written via markdown but auto-generated into TeX-based PDF and a self-hosted website.

I should add more explanations to address a guideline regarding how to contributes to the repo as well as a CI service for auto releasing. Thanks for your reminder.